redbot/cogs/mod/mod.py

```diff
@@ -77,6 +77,7 @@ def __init__(self, bot: Red):
         self.settings.register_user(**self.default_user_settings)
         self.cache: dict = {}
         self.tban_expiry_task = self.bot.loop.create_task(self.check_tempban_expirations())
+        self.init_task = self.bot.loop.create_task(self.initialize())
```

Comment:
This task should be made *after* the cog is added to the bot, otherwise the task might also be created when cog fails to add (e.g. some command name is already taken by other cog)

Just a simple method that you would call from `setup()` after `bot.add_cog(Mod())` would work:
```py
def create_init_task(self):
    self._init_task = asyncio.create_task(self.initialize())
```

Reply:
Updated.